lib/iex/lib/iex/autocomplete.ex

```diff
@@ -297,12 +297,36 @@ defmodule IEx.Autocomplete do
       not ensure_loaded?(mod) ->
         []
       docs = Code.get_docs(mod, :docs) ->
-        exports(mod) |> Enum.reject(&hidden_fun?(&1, docs))
+        (exports(mod) -- get_default_arg_functions_with_doc_false(docs))
+        |> Enum.reject(&hidden_fun?(&1, docs))
       true ->
         exports(mod)
     end
   end
 
+  defp get_default_arg_functions_with_doc_false(docs) do
```

Comment:
Since we are already building this list, we may as well go up to `arity` and get all functions with false and do the `--`. This way we no longer need to check for docs in `hidden_fun?/2`. Basically, change the lines below to:

```elixir
count = Enum.count(...)

Enum.reduce((arity-count)..arity, acc, fn new_arity, new_acc ->
  [{fun_name, new_arity} | new_acc]
end)
```